lib/bitstyles_phoenix/live/dropdown.ex

```diff
@@ -0,0 +1,57 @@
+defmodule BitstylesPhoenix.Live.Dropdown do
+  use BitstylesPhoenix.Component
+
+  alias BitstylesPhoenix.Component.Dropdown, as: RawDropdown
+  alias Phoenix.LiveView.JS
+
+  import Phoenix.LiveView.Utils, only: [random_id: 0]
+
+  @doc """
+  Renders a dropdown component with a button, a menu and options.
+  """
+
+  def ui_dropdown(assigns) do
+    extra = assigns_to_attributes(assigns, [:menu, :button])
+
+    button_assigns = assigns_from_single_slot(assigns, :button, with: :button_extra)
+
+    menu_assigns =
+      assigns_from_single_slot(assigns, :menu,
+        with: fn slot, extra -> get_menu_assigns(slot.id, extra) end,
+        exclude: [:id],
+        default: &get_menu_assigns/0
+      )
+
+    assigns =
+      assigns
+      |> assign(extra: extra)
+      |> assign(button_assigns)
+      |> assign(menu_assigns)
+
+    ~H"""
+    <RawDropdown.ui_dropdown {@extra}>
+      <:button
+        phx-click={JS.toggle(to: "##{@menu_id}",
+                             in: {"is-transitioning", "is-off-screen", "is-on-screen"},
+                             out: {"is-transitioning", "is-on-screen", "is-off-screen"})}
+        aria-controls={@menu_id}
+        {@button_extra}
+        >
+        <%= render_slot(@button) %>
+      </:button>
+      <:menu
+        id={@menu_id}
+        style="display: none"
+        phx-click-away={JS.hide(to: "##{@menu_id}",transition: {"is-transitioning", "is-on-screen", "is-off-screen"})}
+        {@menu_extra}
+        />
+    </RawDropdown.ui_dropdown>
+    """
+  end
+
+  defp get_menu_assigns(), do: get_menu_assigns(random_id(), %{})
+
+  defp get_menu_assigns(nil, extra), do: get_menu_assigns(random_id(), extra)
+
+  defp get_menu_assigns(id, extra), do: [menu_id: id, menu_extra: extra]
```

Comment:
That does not make things easier, since I have to get the `id` from the slot...

Reply:
so if you did

```
assigns_from_single_slot(assigns, :menu, with: :menu_extra)
|> put_random_id()
```

```
put_random_id(menu_assigns) do
  id = get_in(menu_assigns, [:menu_extra, :id]) || random_id()
 [{:menu_id, id} | menu_assigns]
end
```

wouldn't this be more or less the same?

Reply:
it's probably not entirely correct :) but you get the idea, i guess

Reply:
hmm fair enough, although I think this would just be different instead of simpler

Reply:
or maybe you are right, I'll check

Reply:
It does not work, since then we can't make the menu slot optional... Either I would rewrite the `assigns_from_single_slot` completely or I merge away. 

I can still refactor that function when it seems in the way for other components.